File: app/src/main/java/com/example/familymafiaapp/entities/RatingPlayerStats.kt
package com.example.familymafiaapp.entities

data class RatingPlayerStats(
    val seasonId: Int,
    val player: String,
    val ratingCoefficient: Float = 0F,
    val wins: Int = 0,
    val gamesPlayed: Int = 0,
    val winRate: Float = 0F,
    val additionalPoints: Float = 0F,
    val penaltyPoints: Float = 0F,
    val bestMovePoints: Float = 0F,
    val firstKilled: Int = 0,
    val firstKilledCityLost: Int = 0,
    val percentOfDeath: Float = 0F,
    val ciForGame: Float = 0F,
    val ci: Float = 0F,
    val mvp: Float = 0F,
    val winByRole: List<Pair<String, Int>> = emptyList(),
    val gamesForRole: List<Pair<String, Int>> = emptyList(),
    val bestMoveAndAdditionalPointsByRole: List<Pair<String, Float>> = emptyList(),
    val penaltyPointsByRole: List<Pair<String, Float>> = emptyList(),
    val seasonGameLimit: Int = 0,
)

File: lib/models/rating_player_stats.dart
import 'package:family_mafia_app/models/player.dart';
import 'package:freezed_annotation/freezed_annotation.dart';

part 'rating_player_stats.freezed.dart';

@freezed
class RatingPlayerStats with _$RatingPlayerStats {
  const factory RatingPlayerStats({
    required int seasonId,
    required Player player,
    @Default(0.0) double ratingCoefficient,
    @Default(0) int wins,
    @Default(0) int gamesPlayed,
    @Default(0.0) double winRate,
    @Default(0.0) double additionalPoints,
    @Default(0.0) double penaltyPoints,
    @Default(0.0) double bestMovePoints,
    @Default(0) int firstKilled,
    @Default(0) int firstKilledCityLost,
    @Default(0.0) double percentOfDeath,
    @Default(0.0) double ciForGame,
    @Default(0.0) double ci,
    @Default(0.0) double mvp,
    // (roleSheetValue, count)
    @Default([]) List<(String, int)> winByRole,
    @Default([]) List<(String, int)> gamesForRole,
    @Default([]) List<(String, double)> bestMoveAndAdditionalPointsByRole,
    @Default([]) List<(String, double)> penaltyPointsByRole,
    @Default(0) int seasonGameLimit,
    @Default(0.0) double protocolPoints,
    @Default(0) int protocolCorrectGuesses,
    @Default(0) int protocolTotalGuesses,
  }) = _RatingPlayerStats;
}
